diff --git a/src/BottomSheetView.tsx b/src/BottomSheetView.tsx
index cc6f6c9ab4ee9c32897e5fb7b757a781778a01a5..74b5fe58d3256b2540c7117b2bc375d46dd0346e 100644
--- a/src/BottomSheetView.tsx
+++ b/src/BottomSheetView.tsx
@@ -9,6 +9,7 @@ import * as React from 'react';
 import { Platform, StyleSheet } from 'react-native';
 import { SafeAreaProvider } from 'react-native-safe-area-context';
 import { FullWindowOverlay } from 'react-native-screens';
+import { Portal } from "@gorhom/portal"
 import type {
   BottomSheetDescriptorMap,
   BottomSheetNavigationConfig,
@@ -132,7 +133,7 @@ export function BottomSheetView({ state, descriptors }: Props) {
   return (
     <>
       {firstScreen.render()}
-      <Overlay>
+      <Portal hostName='sheets'>
         {shouldRenderProvider.current && (
           <BottomSheetModalProvider>
             {state.routes.slice(1).map((route) => {
@@ -176,7 +177,7 @@ export function BottomSheetView({ state, descriptors }: Props) {
             })}
           </BottomSheetModalProvider>
         )}
-      </Overlay>
+      </Portal>
     </>
   );
 }
